File: trivially/src/main/java/trivially/ep/m5/finalized/Add.java
package trivially.ep.m5.finalized;

import trivially.ep.m5.Exp;

public class Add implements trivially.ep.m5.Add {

    protected Exp left;

    public Exp getLeft() {
        return this.left;
    }

    protected Exp right;

    public Exp getRight() {
        return this.right;
    }

    // Warning: Unsafe casts
    public void setLeft(trivially.ep.m4.Exp left) { this.left = (Exp) left; }
    public void setRight(trivially.ep.m4.Exp right) { this.right = (Exp) right; }

    public Add(Exp left, Exp right) {
        this.left = left;
        this.right = right;
    }
}

File: coco_csharp/CoCo/EP/M5/Finalized/Add.cs
namespace EP.M5.Finalized {
    public class Add : Exp, EP.M5.Add<Exp> {
        private EP.Exp<Exp> left;
        private EP.Exp<Exp> right;

        public Add(EP.Exp<Exp> _left, EP.Exp<Exp> _right) {
            this.left = _left;
            this.right = _right;
        }

        public EP.Exp<Exp> getLeft() { return this.left; }
        public void setLeft(EP.Exp<Exp> _left) { this.left = _left; }

        public EP.Exp<Exp> getRight() { return this.right; }
        public void setRight(EP.Exp<Exp> _right) { this.right = _right; }
    }
}